### PR TITLE
Jira connector refactoring

### PR DESCRIPTION
**Pull request by darshangodase**
## Description

1)Removed adf to markdown conversion for further processing blockgroups data ,now from api response take html and it further parse throught html parser to build blocks (major change) removed all dead code .
2)fixed timestamp issue of jql in incremental sync - converting time based on authenticated user timezone so no data loss or overfetch will happen
3)fixed handling of parent child edges for custom level tickets  that are above epic (epic can have also parent)
4)for notification title populated connector instance name from connector service in connector
5)handling of deletion of records properly with on_records_deleted_cascade this method (this is using for kb and other connectors)
6)removed baseurl from jira personal connector configuration as not required (can use resource level credential)
7)Added notications in jira cloud team and personal connector
8)handled case where if block is child block of another block so there parent_index will be none and and parent_block_index will be their parent block index in further processing
9)In block parser(seperate pasring)html blockgroup further processing not handled fixed that one
10)for large attachment fixed loading whole Jira attachments when streaming ,now it happen in chunks with httpx streaming  
11)issue fetch batch size changed from 50 to 100 to avoid rate limit 
12)added retry mechnism in case of rate limiting 
13)Added concurrency in fetch groups ,fetch roles and to build project recordgroups
14)Reconciliation was not working as every time going new record event so now when record is updated then updaterecord event going so Reconciliation is working now for jira connector
15)fixed comments threading as passing all comments as blockgroup under one comment thread blockgroup by sorting base on created timestamp
16)in reindex record Issue links wiping fixed that issue and then similarly creator,reportor,assignee emails can be none for some users so this for record entity edges will wiped out fixed that issues
17)handling of placeholder records for jira cloud teams and personal connector
18)fix: sync only non inline images and all files
19)update personal connector app name mismatch

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change
- [ ]  in API Request, Response? If yes, documentation is required)
- [x] Performance improvement
- [x] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [x] Unit tests pass
- [x] Integration tests pass
- [x] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [x] **Search capabilities** are working correctly
- [x] **Knowledge search** is functioning properly
- [x] **Connector indexing** is working as expected
- [x] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [x] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * Added dedicated Jira Cloud Personal connector support.
  * Improved Jira Cloud content rendering with inline images, attachments, mentions, and streamed downloads.
  * Connector instances now retain their configured names.

* **Bug Fixes**
  * Improved synchronization reliability with retries, clearer failure notifications, and safer partial-sync handling.
  * Preserved child records during attachment-only deletions.
  * Improved placeholder, deleted-record, permission, and orphan-link handling.
  * Enhanced indexing and block relationships for Jira content.

### CODE REVIEW

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 1]
```diff
@@ -835,61 +313,76 @@ def __init__(
...
             return
         try:
-            if response.status == HttpStatusCode.OK.value:
-                email = (response.json() or {}).get("emailAddress")
-                if email:
-                    self._authenticated_jira_email = email.strip()
+            if response.status != HttpStatusCode.OK.value:
+                return
+            data = response.json() or {}
         except Exception as e:
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:394]: add and handle JsonParsingError or similar for handling exception here

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 2]
```diff
@@ -903,19 +396,29 @@ async def init(self) -> bool:
...
+                severity=NotificationSeverity.ERROR,
+                title=self._notification_title("connection failed"),
+                message=(
+                    f"PipesHub couldn't connect to Jira: {str(e)[:200]}. "
+                    "Verify the connector's credentials and configuration, "
+                    "re-authenticate if needed, then sync again."
+                ),
+                recipient_user_ids=[self.created_by],
+            )
             return False
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: throw exception here in generic Exception handling to propagate it to frontend

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 3]
```diff
@@ -928,82 +431,151 @@ async def _notify_multi_site_ambiguity(self, error: AtlassianMultiSiteError) ->
...
+            await self.notify(
+                type=NotificationType.CONNECTOR_SUCCESS,
+                severity=NotificationSeverity.SUCCESS,
+                title=self._notification_title("sync completed"),
+                message=(
+                    f"Total: {sync_stats['total_synced']} issues "
+                    f"(New: {sync_stats['new_count']}, Updated: {sync_stats['updated_count']})"
+                ),
+                recipient_roles=[NotificationRecipientRole.ADMIN],
+            )
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: remove this notification for success of run sync

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 4]
```diff
@@ -928,82 +431,151 @@ async def _notify_multi_site_ambiguity(self, error: AtlassianMultiSiteError) ->
...
+                    severity=NotificationSeverity.ERROR,
+                    title=self._notification_title("sync failed"),
+                    message=(
+                        f"Issues could not be synced for any of the {failed_count} projects. "
+                        "Run the sync again; if it keeps failing, check the connector's "
+                        "Jira access and configuration."
+                    ),
+                    recipient_roles=[NotificationRecipientRole.ADMIN],
+                )
+                return
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: also send notification if some projects failed

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 5]
```diff
@@ -4740,21 +4159,22 @@ async def test_connection_and_access(self) -> bool:
...
             self.logger.error(f"❌ Connection test failed: {e}")
-            await self.notify(
-                type=NotificationType.CONNECTOR_AUTH_ERROR,
-                severity=NotificationSeverity.ERROR,
-                title=f"Connection failed",
-                message=f"{self.connector_name.value}: {e}",
-                recipient_roles=[NotificationRecipientRole.ADMIN],
-            )
+            await self._notify_connection_test_failed()
             return False
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:4620]: check if its needed to send notification from init and test connection method, as they are synchronous calls, and their errors are shown in frontend if fails

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 6]
```diff
@@ -4451,142 +3932,68 @@ async def _process_issue_blockgroups_for_streaming(self, record: Record) -> byte
...
+                self.logger.debug(f"Skipping large inline image {att_id} ({len(content_bytes)} bytes) — not inlined")
+                cache[att_id] = None
+                return None
 
+            result = f"data:{mime_type};base64,{base64.b64encode(content_bytes).decode('utf-8')}"
         except Exception as e:
-            self.logger.warning(f"⚠️ Error fetching media (id='{media_id}', alt='{media_alt}') for issue {issue_id}: {e}")
-            return None
+            self.logger.warning(f"⚠️ Error fetching inline image attachment {att_id}: {e}")
+            result = None
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:4436]: if images are being inlined then their attachment record should not be created

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 3 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: done

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 1 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:394]: fixed

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 2 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: when return false we throwing exception in router

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 4 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: fixed in this same notification

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 5 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:4620]: removed from init and test connection  and added in run sync

File: backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py [hunk 7]
```diff
@@ -1327,17 +1327,21 @@ async def on_record_deleted(self, record_id: str) -> None:
 
     @retry_on_deadlock()
     async def on_records_deleted_cascade(
-        self, record_ids: list[str], connector_id: str
+        self,
+        record_ids: list[str],
+        connector_id: str,
+        cascade_children: bool = True,
     ) -> dict:
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py:1546]: check if there is a better way to handle record deletion with attachment, as this could be confusing and according to me does not look like a standard way of handling that

File: backend/python/app/connectors/sources/atlassian/core/html_utils.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
+"""Shared helpers for turning Jira/Confluence rendered HTML into indexable blocks.
+
+The Atlassian connectors (Jira Cloud, Jira Data Center) all consume Jira's *rendered*
+HTML (`renderedFields`) for issue descriptions and comments, then hand it to docling for
+indexing. This module centralises the two operations that path needs so the logic is
+written and fixed once:
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/core/html_utils.py]: if this file is not being used for confluence then update this comment

File: backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py [hunk 9]
```diff
@@ -265,12 +245,26 @@ async def run_sync(self) -> None:
                 # ``init()`` returns False on missing/invalid auth config; check
                 # the return so a misconfigured connector raises here instead of
                 # surfacing ``ValueError("DataSource not initialized")`` from
-                # the first datasource call several layers down.
+                # the first datasource call several layers down. Init does not
+                # notify (FE covers setup); notify here for background sync.
                 if not await self.init():
-                    raise RuntimeError(
+                    await self.notify(
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py]: why are we calling init in run_sync? init should have already been executed by the time run sync is called, or should throw an error

File: backend/python/app/connectors/sources/atlassian/core/html_utils.py (hunk 8 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/core/html_utils.py]: fixed

File: backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py (hunk 9 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py]: fixed

Comment (shrikantNemiwal1) [reply backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py]: i dont see any changes for the fix, have you pushed your changes?

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 2 above)
Comment (shrikantNemiwal1) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: but it will not propagate the error we get here, we need to propagate the error message as well to the user

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 10]
```diff
@@ -928,82 +425,157 @@ async def _notify_multi_site_ambiguity(self, error: AtlassianMultiSiteError) ->
...
+
+    async def run_sync(self) -> None:
+        """Main sync flow: users → groups → projects/roles → issues → deletions."""
+        try:
+            # 1. Ensure client is ready. Init itself does not notify (FE shows setup
+            # errors); notify here for background sync when auth/config is broken.
+            if not self.data_source:
+                if not await self.init():
+                    await self.notify(
+                        type=NotificationType.CONNECTOR_AUTH_ERROR,
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: same here as personal connector, dont call init here

File: backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py (hunk 7 above)
Comment (darshangodase) [reply backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py:1546]: fixed

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 11]
```diff
@@ -928,82 +425,144 @@ async def _notify_multi_site_ambiguity(self, error: AtlassianMultiSiteError) ->
...
+            await self._sync_project_lead_roles(raw_projects, jira_users)
+            await self.data_entities_processor.on_new_record_groups(projects)
+
+            # 7. Sync issues (incremental via checkpoint), then deletions
+            last_sync_time = await self._get_issues_sync_checkpoint()
+            sync_stats = await self._sync_all_project_issues(projects, jira_users, last_sync_time)
+            await self._update_issues_sync_checkpoint(sync_stats, len(projects))
+            await self._handle_issue_deletions(last_sync_time)
+
+            # 8. Outcome: notify when any project failed issue sync (include keys)
```
Comment (shrikantNemiwal1) [inline backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: handle placeholder records here after sync like how confluence connector does, so that hierarchy is maintained even though records are not synced

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 6 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:4436]: fixed,skipping inline images from syncing

File: backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py (hunk 9 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud_personal/connector.py]: image
can u check again

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 10 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: fixed this also, check again

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 2 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: ok then willl pass throught connectorinitError which propogate actual error message to user

File: backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py [hunk 12]
```diff
@@ -774,19 +797,23 @@ async def mock_audit(**kwargs):
...
         ds.get_audit_records = AsyncMock(return_value=_resp(500, {}))
         connector._get_fresh_datasource = AsyncMock(return_value=ds)
+        # A non-OK audit page notifies (missing admin permission) and reports ok=False.
+        connector.notify = AsyncMock()
 
-        result = await connector._fetch_deleted_issues_from_audit("2024-01-01T00:00:00.000Z", "2024-01-31T00:00:00.000Z")
-        assert result == []
+        keys, ok = await connector._fetch_deleted_issues_from_audit("2024-01-01T00:00:00.000Z", "2024-01-31T00:00:00.000Z")
+        assert keys == []
+        assert ok is False
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py:817]: fixed

File: backend/python/tests/unit/connectors/sources/test_jira_connector.py [hunk 13]
```diff
@@ -2986,16 +2179,19 @@ async def test_grants_fetch_failure(self):
...
     @pytest.mark.asyncio
-    async def test_exception_returns_empty(self):
+    async def test_exception_returns_none(self):
         connector = _make_connector()
         connector._get_fresh_datasource = AsyncMock(side_effect=Exception("err"))
 
+        # Couldn't determine the scheme at all -> None (skip, preserve ACL).
         permissions = await connector._fetch_project_permission_scheme("PROJ")
-        assert permissions == []
+        assert permissions is None
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_connector.py]: fixed

File: backend/python/tests/unit/connectors/sources/test_jira_connector_workflow.py [hunk 14]
```diff
@@ -839,83 +430,67 @@ async def test_handle_deleted_issue_still_exists(self):
...
-        child.external_record_id = "child-ext-1"
-        tx.get_records_by_parent = AsyncMock(side_effect=[
-            [child],  # first call for subtasks
-            [],  # nested subtasks of child
-            [],  # attachments of child
-        ])
-
-        result = await connector._delete_issue_children("parent-ext-1", RecordType.TICKET, tx)
-        assert result == 1
+        dep.on_records_deleted_with_attachments.assert_not_awaited()
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_connector_workflow.py]: fixed

File: backend/python/tests/unit/connectors/sources/test_jira_cloud_connector.py [hunk 15]
```diff
@@ -829,107 +834,67 @@ async def test_non_epic_deletion_with_children(self):
...
-        tx_store = AsyncMock()
-        tx_store.get_records_by_parent = AsyncMock(side_effect=Exception("db error"))
+        mock_tx, mock_tx_store = _make_tx_store(None)
+        connector.data_store_provider.transaction = MagicMock(return_value=mock_tx)
 
-        count = await connector._delete_issue_children("parent-1", RecordType.FILE, tx_store)
-        assert count == 0
+        with pytest.raises(Exception, match="unconfirmed"):
+            await connector._handle_deleted_issue("PROJ-1")
+        connector.data_entities_processor.on_records_deleted_with_attachments.assert_not_awaited()
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_cloud_connector.py]: fixed

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 16]
```diff
@@ -4782,17 +4576,54 @@ async def cleanup(self) -> None:
...
+
     async def stream_record(self, record: Record) -> StreamingResponse:
         """
         Stream record content (issue, comment, or attachment).
         """
+        if record.is_placeholder:
+            raise ValueError(
+                f"Cannot stream placeholder record {record.external_record_id}: "
+                "it is a stub for an out-of-scope ancestor and has no content"
+            )
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:4697]: done

File: backend/python/app/sources/external/jira/jira.py [hunk 17]
```diff
@@ -5444,8 +5475,8 @@ async def bulk_fetch_issues(
             method='POST',
             url=url,
             headers=_as_str_dict(_headers),
-            path_params=_as_str_dict(_path),
-            query_params=_as_str_dict(_query),
+            path=_as_str_dict(_path),
+            query=_as_str_dict(_query),
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/sources/external/jira/jira.py:5479]: handle wherever required

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 18]
```diff
@@ -112,563 +119,52 @@
...
+METADATA_FETCH_CONCURRENCY: int = 5
 
 # JQL query constants
 ISSUE_SEARCH_FIELDS: list[str] = [
     "summary", "description", "status", "priority",
     "creator", "reporter", "assignee", "created", "updated",
-    "issuetype", "project", "parent", "attachment", "security",
+    "issuetype", "project", "parent", "attachment",
     "issuelinks"
 ]
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:193]: fixed

File: backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py [hunk 19]
```diff
@@ -1355,17 +1355,45 @@ async def on_record_deleted(self, record_id: str) -> None:
...
+                "total_requested": 0,
+                "successfully_deleted": 0,
+                "failed_count": 0,
+            }
+        async with self.data_store_provider.transaction() as tx_store:
+            result = await tx_store.delete_records_recursive(
+                record_ids, connector_id, cascade_children=True,
+            )
+        await self._publish_delete_events((result or {}).get("eventData"))
+        return result
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/core/base/data_processor/data_source_entities_processor.py:1589]: done

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 20]
```diff
@@ -1982,197 +1625,346 @@ async def _fetch_project_permission_scheme(
...
 
-            # Step 2: Get all permission grants in this scheme
-            grants_response = await datasource.get_permission_scheme_grants(
-                schemeId=scheme_id,
-                expand="all"
+            # Step 2: Get all permission grants in this scheme (transport + 429 retry)
+            grants_response = await self._call_with_retry(
+                lambda ds: ds.get_permission_scheme_grants(schemeId=scheme_id, expand="all"),
+                ctx=f"permission grants (scheme {scheme_id}) for {project_key}",
             )
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:1705]: done

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py (hunk 11 above)
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py]: done

File: backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py [hunk 21]
```diff
@@ -1248,11 +1286,1030 @@ async def test_anyone_holder_permission(self):
...
+
+    @pytest.mark.asyncio
+    async def test_attachment_fetch_exception_returns_none(self):
+        connector = _make_connector()
+        connector.indexing_filters = MagicMock()
+        connector._get_fresh_datasource = AsyncMock(side_effect=RuntimeError("boom"))
+
+        record = _stub_record("attachment_5")
+        record.record_type = RecordType.FILE
+        assert await connector._check_and_fetch_updated_attachment(record) is None
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py:2133]: minor,not required

File: backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py [hunk 22]
```diff
@@ -1248,11 +1286,1030 @@ async def test_anyone_holder_permission(self):
...
+    @pytest.mark.asyncio
+    async def test_fetch_groups_non_ok_flags_failure(self):
+        connector = _make_connector()
+        ds = MagicMock()
+        ds.get_groups_paginated = AsyncMock(return_value=_resp(500, {}))
+        connector._get_fresh_datasource = AsyncMock(return_value=ds)
+        connector.data_source = MagicMock()
+
+        groups, forbidden = await connector._fetch_groups()
+        assert groups == []
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py:2259]: minor,not required

File: backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py [hunk 23]
```diff
@@ -1248,11 +1286,1030 @@ async def test_anyone_holder_permission(self):
...
+
+    def test_non_ok_status_returns_early(self):
+        connector = _make_connector()
+        connector._cache_authenticated_jira_profile(_resp(500, {}))
+
+    def test_unknown_timezone_falls_back_to_utc(self):
+        connector = _make_connector()
+        connector._cache_authenticated_jira_profile(
+            _resp(200, {"emailAddress": "a@b.com", "timeZone": "Not/AZone"})
+        )
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py:1926]: minor,not required

File: backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py [hunk 24]
```diff
@@ -1248,11 +1286,273 @@ async def test_anyone_holder_permission(self):
...
     @pytest.mark.asyncio
-    async def test_scheme_fetch_failure_returns_empty(self):
+    async def test_scheme_fetch_failure_returns_none(self):
         connector = _make_connector()
         ds = MagicMock()
         ds.get_assigned_permission_scheme = AsyncMock(return_value=_resp(500, {}))
         connector._get_fresh_datasource = AsyncMock(return_value=ds)
 
+        # Transient 5xx -> None; the caller syncs the RecordGroup with an empty ACL
+        # (returning [] would overwrite the ACL and hide the project from everyone).
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_deep_sync.py:1297]: minor,not required

File: backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py [hunk 25]
```diff
@@ -3364,159 +3210,287 @@ async def _build_issue_records(
...
+                continue
+            mime = (attachment.get("mimeType") or "").lower()
+            if not mime.startswith("image/"):
+                continue
+            # Only images small enough to actually be inlined are suppressed. An oversized
+            # one is dropped to alt text at stream time, so without this it would end up
+            # with neither a FileRecord nor indexed content.
+            size_bytes = int(attachment.get("size") or 0)
+            if size_bytes > MAX_INLINE_IMAGE_BYTES:
+                continue
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/sources/atlassian/jira_cloud/connector.py:3532]: validated ,no changes required

File: backend/python/tests/unit/connectors/sources/test_jira_connector.py [hunk 26]
```diff
@@ -2530,140 +1803,73 @@ async def test_error_continues(self):
...
+    @pytest.mark.asyncio
+    async def test_exception_returns_none(self):
+        connector = _make_connector()
+        connector._get_fresh_datasource = AsyncMock(side_effect=Exception("boom"))
+        cache = {}
+        result = await connector._fetch_attachment_as_base64(
+            {"id": "42", "mimeType": "image/png", "size": 10}, cache
+        )
         assert result is None
+        assert cache["42"] is None
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/tests/unit/connectors/sources/test_jira_connector.py:1857]: Keeping these each covers a different branch, not redundant happy path copies

**Review by abhishek-pipeshub — APPROVED** [review]
